colibri/src/icons.rs

```diff
@@ -30,23 +41,29 @@ fn get_headers(extension: &str) -> Result<[(&'static str, &'static str); 2], Fil
     Ok([("Content-Type", content_type), ("mimetype", content_type)])
 }
 
-fn extract_chain_id_and_address(identifier: &str) -> Option<(u64, String)> {
+fn parse_asset_identifier(identifier: &str) -> Option<(u64, String, Option<String>)> {
     let parts: Vec<&str> = identifier.split('/').collect();
-    if parts.len() != 2 {
+    if parts.len() < 2 {
         return None;
     }
 
-    let chain_parts: Vec<&str> = parts[0].split(':').collect();
-    let address_parts: Vec<&str> = parts[1].split(':').collect();
-
-    if chain_parts.len() != 2 || address_parts.len() != 2 {
+    // Parse chain ID
+    let chain_parts: Vec<&str> = parts[0].splitn(2, ':').collect();
+    if chain_parts.len() != 2 || chain_parts[0] != "eip155" {
         return None;
     }
-
     let chain_id = chain_parts[1].parse::<u64>().ok()?;
-    let address = address_parts[1].to_string();
 
-    Some((chain_id, address))
+    // Parse asset type and contract address
+    let asset_parts: Vec<&str> = parts[1].splitn(2, ':').collect();
+    if asset_parts.len() != 2 {
+        return None;
+    }
+    let contract_address = asset_parts[1].to_string();
```

Comment:
can't this break if somehow the id is wrong and it is missing parts? Is it possible to access with a get method that allows you to catch errors?